django/contrib/admin/static/admin/js/actions.js

```diff
@@ -142,28 +142,70 @@
             return filtered;
         };
 
-        Array.from(document.getElementById('result_list').tBodies).forEach(function(el) {
-            el.addEventListener('change', function(event) {
+        if (!options.inline) {
+            document
+                .getElementById(options.allToggleId)
+                .addEventListener('click', function(event) {
+                    checker(actionCheckboxes, options, this.checked);
+                    updateCounter(actionCheckboxes, options);
+                });
+
+            Array.from(document.getElementById('result_list').tBodies).forEach(
+                function(el) {
+                    el.addEventListener('change', function(event) {
+                        const target = event.target;
+                        if (target.classList.contains('action-select')) {
+                            const checkboxes = affectedCheckboxes(target, shiftPressed);
+                            checker(checkboxes, options, target.checked);
+                            updateCounter(actionCheckboxes, options);
+                            lastChecked = target;
+                        } else {
+                            list_editable_changed = true;
+                        }
+                    });
+                });
+
+            document.querySelector('#changelist-form button[name=index]').addEventListener('click', function(event) {
+                if (list_editable_changed) {
+                    const confirmed = confirm(gettext("You have unsaved changes on individual editable fields. If you run an action, your unsaved changes will be lost."));
+                    if (!confirmed) {
+                        event.preventDefault();
+                    }
+                }
+            });
+        }
+
+
+        if (options.inline) {
+            const handleCheckboxChange = (event) => {
                 const target = event.target;
-                if (target.classList.contains('action-select')) {
+
+                if (!lastChecked || !target.name.endsWith('-DELETE')) {
+                    lastChecked = target;
+                    return;
+                }
+
+                if (lastChecked.name.slice(0, -9) === target.name.slice(0, -9)) {
```

Comment:
This is a bit magic, can you explain what you're doing here?

Reply:
It's checking if both check boxes are in the same form. Without this, it's possible to toggle check boxes with different inline admins which shouldn't be.